src/chessboard/components/Squares.tsx

```diff
@@ -18,6 +19,38 @@ export function Squares() {
     showBoardNotation,
   } = useChessboard();
 
+  const premovesHistory: any[] = useMemo(() => {
```

Comment:
maybe this make sense to be upped to the `chessboard-context` state, and keep all the premoves in this format

Reply:
```js  
premoves: [
    {
      sourceSq: "f1",
      targetSq: "e2",
      piece: "wB",
    },
    {
      sourceSq: "c2",
      targetSq: "d3",
      piece: "wP",
    },
    {
      sourceSq: "e2",
      targetSq: "d3",
      piece: "wB",
    },
    {
      sourceSq: "g1",
      targetSq: "f3",
      piece: "wN",
    },
    {
      sourceSq: "f3",
      targetSq: "g1",
      piece: "wN",
    },
    {
      sourceSq: "g1",
      targetSq: "f3",
      piece: "wN",
    },
  ],
```

Reply:
```js
premovesHistory: [
    {
      piece: "wN",
      premovesRoute: [
        {
          sourceSq: "g1",
          targetSq: "f3",
        },
        {
          sourceSq: "f3",
          targetSq: "g1",
        },
        {
          sourceSq: "g1",
          targetSq: "f3",
        },
      ],
      index: 5,
    },
    {
      piece: "wB",
      premovesRoute: [
        {
          sourceSq: "f1",
          targetSq: "e2",
        },
        {
          sourceSq: "e2",
          targetSq: "d3",
        },
      ],
      index: 2,
    },
    {
      piece: "wP",
      premovesRoute: [
        {
          sourceSq: "c2",
          targetSq: "d3",
        },
      ],
      index: 1,
    },
  ],
```